migration/handler.go

```diff
@@ -0,0 +1,52 @@
+package migration
+
+import (
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/x"
+)
+
+// RegisterRoutes registers handlers for feedlist message processing.
+func RegisterRoutes(r weave.Registry, auth x.Authenticator) {
+	bucket := NewSchemaBucket()
+	r.Handle(pathUpgradeSchemaMsg, &upgradeSchemaHandler{bucket: bucket})
+}
+
+type upgradeSchemaHandler struct {
+	bucket *SchemaBucket
+}
+
+func (h *upgradeSchemaHandler) Check(ctx weave.Context, db weave.KVStore, tx weave.Tx) (*weave.CheckResult, error) {
+	if _, err := h.validate(ctx, db, tx); err != nil {
+		return nil, err
+	}
+	return &weave.CheckResult{}, nil
+}
+
+func (h *upgradeSchemaHandler) Deliver(ctx weave.Context, db weave.KVStore, tx weave.Tx) (*weave.DeliverResult, error) {
+	msg, err := h.validate(ctx, db, tx)
+	if err != nil {
+		return nil, err
+	}
+
+	ver, err := h.bucket.CurrentSchema(db, msg.Pkg)
+	if err != nil && !errors.ErrNotFound.Is(err) {
+		return nil, errors.Wrap(err, "current schema version")
+	}
+
+	schema := Schema{Pkg: msg.Pkg, Version: ver + 1}
+	obj, err := h.bucket.Create(db, &schema)
+	if err != nil {
+		return nil, errors.Wrap(err, "create schema version")
+	}
+
+	return &weave.DeliverResult{Data: obj.Key()}, nil
+}
+
+func (h *upgradeSchemaHandler) validate(ctx weave.Context, db weave.KVStore, tx weave.Tx) (*UpgradeSchemaMsg, error) {
```

Comment:
Nice. The handler code is looking shorter and shorter, less boilerplate.

However, we really do need some permission here, who can do the upgrade. Can you path an Address in the constructor of the Handler and enforce that here? It can be a more generic `gconf` solution later, but at least a placeholder for such authorization logic.

Reply:
Done in https://github.com/iov-one/weave/pull/550/commits/bf6175479febe71e130d6a12be63383d2efa2d18